upnp/src/genlib/net/http/httpreadwrite.c

```diff
@@ -505,7 +505,7 @@ int http_SendMessage(SOCKINFO *info, int *TimeOut, const char *fmt, ...)
 				if (Instr) {
 					int nr;
 					size_t n = amount_to_be_read >= Data_Buf_Size ?
-					    	Data_Buf_Size : amount_to_be_read;
+						Data_Buf_Size : (size_t)amount_to_be_read;
```

Comment:
This cast has no effect. If amount_to_be_read is too big for n, it is discarded with and without the cast. I'd say, don't add it.

Reply:
Without the cast is valid C. Only C++ is picky here. Ack, there are many useless casts. So if nobody volunteers to remove them, let's at least stop to introduce new ones.

Reply:
Not saying it's not valid, just that I'd like a warning.
Out of curiosity, do you know why C (and C++ actually) don't warn (even with -Wall) in this case ? This is really weird, we get many not so useful warnings, and for this, which is actually dangerous and a source of bugs nothing ? The only case where I get a warning is with a c++11 initializer: 
size_t x{someofft};

I don't understand the rationale.